test/unit/sql/cte.test.js

```diff
@@ -0,0 +1,332 @@
+'use strict';
+
+
+/* jshint -W110 */
+var Support   = require(__dirname + '/../support')
+  , DataTypes = require(__dirname + '/../../../lib/data-types')
+  , expectsql = Support.expectsql
+  , current   = Support.sequelize
+  , sql       = current.dialect.QueryGenerator;
+  
+describe(Support.getTestDialectTeaser('SQL'), function () {
+  if (current.dialect.supports.cteQueries) {
+    describe('CTE', function () {
+      var User = current.define('user', { amount: DataTypes.INTEGER});
+      var Project = current.define('project', { name: DataTypes.STRING, budget: DataTypes.INTEGER});
+      
+      User.hasOne(User, { as: 'report' });
+      User.hasMany(Project, { as: 'assigned'});
+
+      var selectTest = function(options, expected) {
+        
+        // just for testing
+        options.model = User;
+
+        current.Model._conformOptions.call(User, options, User);
+        if (options.cte) {
+          current.Model._validateCTEElements.call(User, options);
+        }
+        expectsql(sql.selectQuery(User.getTableName(), options, User), expected);
+      };
+
+
+      it('automatically connects to single CTE', function () {
```

Comment:
would like to see a unit test of a simple case where the CTE is simply used as a subquery helper of sorts for filtering.